AssetRipper.GUI/ConsoleApp.cs

```diff
@@ -1,126 +1,179 @@
-﻿using AssetRipper.Core.IO.MultiFile;
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.CommandLine.Parsing;
+using System.CommandLine.Builder;
+
 using AssetRipper.Core.Logging;
-using AssetRipper.Core.Utils;
 using AssetRipper.Library;
-using CommandLine;
-using System.Collections.Generic;
-using System.IO;
 
 namespace AssetRipper.GUI
 {
 	internal static class ConsoleApp
 	{
-		private const string DefaultLogFileName = "AssetRipper.log";
-
-		internal class Options
+		private static void DoRipping(Ripper ripper, List<string> inputPaths, string outputPath, bool verbose, string logPath)
 		{
-			[Value(0, Required = true, HelpText = "Input files or directory to export.")]
-			public IReadOnlyList<string> FilesToExport { get; set; }
+			Logger.AllowVerbose = verbose;
+			Logger.Add(new ConsoleLogger(false));
 
-			[Option('o', "output", HelpText = "Directory to export to. Will be cleared if already exists.")]
-			public DirectoryInfo OutputDirectory { get; set; }
+			// Do not log to a file if the logging target is null. It should be set from ConsoleOptions.DefaultLogFileName
+			// if the user didn't specify one.
+			if (string.IsNullOrEmpty(logPath))
+				return;
```

Comment:
A log file should be created always.

Reply:
A log file is always created, as the default value is set to "AssetRipper.log". This check is to prevent throwing in case that variable is somehow unset between parsing the arguments and actually accessing its value. It could be due to a bug, or anything else, since System.CommandLine is not marked as stable yet.  

Primarily, it acts as a guard because filePath may technically be an empty string.